libdnf/base/goal_elements.cpp

```diff
@@ -38,22 +38,24 @@ bool GoalJobSettings::resolve_strict(const libdnf::ConfigMain & cfg_main) {
             resolved = GoalUsedSetting::USED_FALSE;
             break;
     }
-    if (resolved == GoalUsedSetting::UNUSED) {
-        throw LogicError("Invalid 'strict' value in GoalJobSettings");
-    }
-    if (used_strict != GoalUsedSetting::UNUSED && resolved != used_strict) {
-        throw LogicError("GoalJobSettings::resolve_strict: 'strict' is already set to a different value");
-    }
+
+    // TODO this cannot happen, code above always sets USED_TRUE or USED_FALSE
+    libdnf_assert(resolved != GoalUsedSetting::UNUSED, "Invalid 'strict' value in GoalJobSettings");
```

Comment:
But the assert is meant for outside conditions, by extension of your logic you could write:
```C++
in x = 5;
assert(x == 5);
```
Because anyone could change the value of x. If anyone changes the switch, they either make an intentional change or they introduce a bug, just as they can do with any other piece of code. Even if we litter our code with asserts all over it's not going to help.

I find this assert redundant and misleading because a reader would try to figure out what this assert is for only to find out the situation cannot happen (just like I had to do). If it can't happen, I'd like to remove it (and the other similar asserts).